medicines/doc-index-updater/src/delete_manager/mod.rs

```diff
@@ -25,38 +30,57 @@ pub async fn delete_service_worker(
+        match try_process_from_queue(&mut delete_client, &search_client, &storage_client).await {
+            Ok(status) => match status {
+                FileProcessStatus::Success(job_id) => {
+                    let _ = state_manager.set_status(job_id, JobStatus::Done).await?;
+                }
+                FileProcessStatus::NothingToProcess => {}
+            },
+            Err(e) => {
+                tracing::error!("{:?}", e);
+            }
+        }
+        delay_for(time_to_wait).await;
+    }
+}
 
-        match retrieved_result {
-            Ok(retrieval) => {
-                let message = retrieval.message.clone();
-                tracing::info!("{:?} message receive!", message);
-                let blob_name =
-                    get_blob_name_from_content_id(&search_client, message.document_content_id)
-                        .await?;
-                delete_from_index(&search_client, &blob_name).await?;
-                delete_blob(&storage_client, &storage_container_name, &blob_name)
-                    .await
-                    .map_err(|e| {
-                        tracing::error!("{:?}", e);
-                        anyhow!("Couldn't delete blob {}", &blob_name)
-                    })?;
+async fn try_process_from_queue(
+    delete_client: &mut DocIndexUpdaterQueue,
+    search_client: &search_client::AzureSearchClient,
+    storage_client: &azure_sdk_storage_core::prelude::Client,
+) -> Result<FileProcessStatus, anyhow::Error> {
+    tracing::info!("Checking for delete messages");
+    let retrieved_result: Result<RetrievedMessage<DeleteMessage>, RetrieveFromQueueError> =
+        delete_client.receive().await;
 
-                retrieval.remove().await?;
-                // TODO: Notify state manager
-            }
-            Err(azure_error) => tracing::warn!("Azure error! {:?}", azure_error),
-        };
+    if let Ok(retrieval) = retrieved_result {
+        let message = retrieval.message.clone();
+        tracing::info!("{:?} message receive!", message);
+        let storage_container_name = std::env::var("STORAGE_CONTAINER")?;
+        let blob_name =
+            get_blob_name_from_content_id(message.document_content_id.clone(), &search_client).await?;
+        delete_from_index(&search_client, &blob_name).await?;
+        delete_blob(&storage_client, &storage_container_name, &blob_name)
+            .await
+            .map_err(|e| {
+                tracing::error!("{:?}", e);
+                anyhow!("Couldn't delete blob {}", &blob_name)
+            })?;
 
-        delay_for(Duration::from_secs(10)).await;
+        retrieval.remove().await?;
+        Ok(FileProcessStatus::Success(message.job_id))
+    } else {
+        Ok(FileProcessStatus::NothingToProcess)
     }
 }
 
-pub async fn get_blob_name_from_content_id(
-    search_client: &search_client::AzureSearchClient,
-    content_id: String,
-) -> Result<String, anyhow::Error> {
+enum FileProcessStatus {
```

Comment:
Can this be in `models.rs` and shared by both managers?

Reply:
Happy for this optimisation to be made in the ticket to update state manager for create service, but may as well do it now, I guess!